### ISSUE TITLE
protect `clientID_` setter in server?

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#3272]
As it is right now, setting a server's clientID is possible at any moment:

```supercollider
x = Server(\test);
x.options.maxLogins = 8;
x.boot
// a little later
x.clientID = 3;

```

this returns a rather nasty looking error, 

```
ERROR: Primitive '_BasicPut' failed.
Index out of range.
RECEIVER:
Instance of Array {    (0x10cfbd080, gc=B4, fmt=01, flg=00, set=0B)
  indexed slots [2048]
      0 : nil
      1 : nil
      2 : nil
      3 : nil
…
```

Since `clientID` is supposed to be a private message (https://github.com/supercollider/supercollider/blob/develop/SCClassLibrary/Common/Control/Server.sc#L387), shouldn't it be called something like `prSetClientID`? 

The following could be done:
```
prSetClientID { |val|
		var failstr = "Server % couldn't set clientID to: % - %. clientID is still %.";
          <etc.>
}

clientID_ { |val|
		if(this.serverRunning) { 
				"cannot set the client ID of a running server".warn; ^this 
		} {
				this.prSetClientID(val)
		}
}
```

**Comment by adcxyz** [comment]
Yes, since ServerStatusWatcher, clientID is open, and options.maxLogins has always been open to change at any time. This opens the possibility to have wrong allocator ranges when remaking them.

clientID should best be locked after the server notify response until next reboot, and ideally maxLogins should be locked too. Even better would be to keep a locked copy of all other fixed settings that scsynth was booted with, so the client really knows the config of the current scsynth.
That seemed to be too much for 3.9.0, so I did not code that up yet. 

I guess the best option now is to provide locking for clientID and maxLogins, ok?
Will write a branch with that ASAP, best adc

**Comment by adcxyz** [comment]
Actually, since clientID is now locked when scsynth is running, we could simplify away the special case of ```userSpecifiedClientID```:
when the server process is not running, you can change clientID as you like (constrained by maxLogins only);
when the server is booted, it will try to get that clientID if possible, or else return a valid one;
when running, you cant change clientID.

**Comment by telephon** [comment]
yes, this is a good idea.
